ci/bumper_service.ts

```diff
@@ -0,0 +1,275 @@
+
+interface Versions {
+  latest: string; // "v1.2.3"
+  versions: string[]; // ["v1.2.3", "v1.2.2", ...[
+}
+
+/**
+ * Used to update version strings as part of the bumper CI process when new
+ * Deno, Deno Std, and Drash Land module version are released.
+ */
+export class BumperService {
+  /**
+   * A property to hold Deno.args.
+   */
+  protected args: string[];
+
+  /**
+   * A property to determine whether or not the .bump() method should bump for
+   * pre-release.
+   */
+  protected is_for_pre_release = false;
+
+  /**
+   * A list of latest versions. This object should contain (at the very least):
+   *     - This module's latest version
+   *     - Deno's latest version
+   *     - Deno Std's latest version
+   */
+  protected latest_versions: { [key: string]: string } = {};
+
+  /**
+   * The name of the module using this class.
+   */
+  protected module_name: string;
+
+  /**
+   * A property to hold the Deno.args parsed into key-value pairs.
+   */
+  protected parsed_args: ParsedArgs;
+
+  //////////////////////////////////////////////////////////////////////////////
+  // FILE MARKER - CONTRUCTOR //////////////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * Construct an object of this class.
+   *
+   * @param moduleName - The name of the module using this class.
+   */
+  constructor(moduleName: string, args: string[]) {
+    this.module_name = moduleName;
+
+    // Make a copy of the args in case they're readonly, which means we can't
+    // mutate them. We want them mutable just in case.
+    this.args = args.slice();
+
+    // Parse the arguments into key-value pairs
+    this.parsed_args = this.getParsedArgs();
+
+    // Are we using this class to bump versions for pre-release?
+    if (this.parsed_args.branch) {
+      this.is_for_pre_release = true;
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // FILE MARKER - METHODS - PUBLIC ////////////////////////////////////////////
+  //////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * Bump all occurances of Deno and Deno Std versions in the filesystem.
+   *
+   * @example
+   * const files = [
+   *   {
+   *     filename: "./README.md",
+   *     replaceTheRegex: /drash@v[0-9.]+[0-9.]+[0-9]/g,
+   *     replaceWith: "drash@v{{ latestDrashRelease }}"
+   *   }
+   * ]
+   *
+   *   - Replace drash versions; use `replaceWith: "{{ latestDrashVersion }}"`
+   *   - Replace std versions; use `...: "{{ latestStdVersion }}"`
+   *   - Replace deno versions; use `...: "{{ latestDenoVersion }}"`
+   *
+   * All versions do not have the `v`, so add them yourself
+   *
+   * @param files - Files to replace content in and re-write to fs
+   * @param write - Should this method write to the filesystem? Defaults to
+   * true.
+   */
+  public async bump(files: File[], write: boolean = true): Promise<string[]> {
```

Comment:
method doesn't seem to replace drash versions? or is that not a thing in this method anymore?

Reply:
is there a reason why it should? i was under the impression that the bump for pre-release is the one that handles updating drash versions.